src/main/java/winstone/Launcher.java

```diff
@@ -229,9 +231,19 @@ private void writePortToFileIfNeeded() throws IOException {
                     Path portFile = Paths.get(portFileName);
                     Path portDir = portFile.getParent();
                     Files.createDirectories(portDir);
-                    try (BufferedWriter writer = Files.newBufferedWriter(portFile, StandardCharsets.UTF_8)) {
+                    Path tmpPath = Files.createTempFile(portDir, "temp", "");
+                    try (BufferedWriter writer = Files.newBufferedWriter(tmpPath, StandardCharsets.UTF_8)) {
                         writer.write(Integer.toString(port));
                     }
```

Comment:
Optional feedback for your consideration: This repository is using Java 11, so we have access to new `Files` methods! That allows us to reduce these three lines to one, which makes it easier to quickly read the code:

```suggestion
                    Files.writeString(tmpPath, Integer.toString(port));
```

Not a blocking comment, since the old `BufferedWriter` approach is pre-existing code that you did not write.